lib/codecs/MultibaseEncoder.js

```diff
@@ -10,14 +10,22 @@ import {decode as decodeBase58} from 'base58-universal';
 // this class is used to encode a multibase encoded value in CBOR-LD, which
 // actually means transforming a multibase-encoded string to bytes
 export class MultibaseEncoder extends CborldEncoder {
-  constructor({value} = {}) {
+  constructor({value, compressionMap} = {}) {
     super();
     this.value = value;
+    this.compressionMap = compressionMap;
   }
 
   encode() {
     const {value} = this;
 
+    // use compression entry for this string if it exists
+
+    if(this.compressionMap.has(value)) {
+      const id = this.compressionMap.get(value);
```

Comment:
EDIT: Read this, but don't use it, use the other suggestion.

Hit the map just once:

```suggestion
    const id = this.compressionMap.get(value);
    if(id !== undefined) {
```

Reply:
But this whole thing looks weird. It seems like, instead, this should be executing in `createEncoder` and passing the `id` (if found) to the constructed decoder -- or falling through to the `value[0]` check below if `id` is undefined.